File: Python/practical/22.py
# Program to check armstrong number

numStr = input("Enter the number: ")

numList = list(numStr)
numList = [int(n) for n in numList]

cubeList = [n ** 3 for n in numList]

if numStr == str(sum(cubeList)):
    print(f"{numStr} is an Armstrong number")
else:
    print(f"{numStr} is not an Armstrong number")

File: C/SEM I/Practical/Done/22.c
/* Write a program in C to accept an integer number and to check whether it is Armstrong Number or not */

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>
#include <math.h>

int count = 0;

int main(void) {
	int rep = 1;

	while (rep == 1) {
		int num, cubeSum = 0, i, a;

		printf("Enter the number: ");
		scanf("%d", & num);

		for (i = num; i > 0; i /= 10) {
			a = (i%10)*(i%10)*(i%10);
			cubeSum += a;
		}

		if(cubeSum==num){
			printf("\nThe number is an Armstrong Number\n");
		}else{
			printf("\nThe number is not an Armstrong Number\n");
		}

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 50) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		}

	}

	printf("Normal END\n");

	getch();
	return 0;
}